### PR TITLE
Issue/type untangling 2

### PR DESCRIPTION
**Pull request by wouterdb**
# Description

Make type tangle less bad:

-   move resource id string type to types module
-  make resources.ID less prone to loops by making resources depend on less
   -  make inmanta.ast contain all exceptions of the compiler
   -  make inmanta.ast import very little else.

Successor of #8466 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] Attached issue to pull request
- [x] Changelog entry
- [X] Type annotations are present
- [x] Code is clear and sufficiently documented
- [x] No (preventable) type errors (check using make mypy or make mypy-diff)
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [X] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )
- [ ] If this PR fixes a race condition in the test suite, also push the fix to the relevant stable branche(s) (see [test-fixes](https://internal.inmanta.com/development/core/tasks/build-master.html#test-fixes) for more info)

### CODE REVIEW

File: src/inmanta/execute/proxy.py [hunk 1]
```diff
@@ -18,9 +18,10 @@
 
 from collections.abc import Iterable, Mapping, Sequence
 from copy import copy
-from typing import Callable, Optional, Union
+from typing import Callable, Union
 
-from inmanta.ast import NotFoundException, RuntimeException
+# Keep UnsetException, UnknownException and AttributeNotFound in place for backward compat with <iso8
+from inmanta.ast import AttributeNotFound, NotFoundException, RuntimeException, UnknownException
```
Comment (Hugo-Inmanta) [inline src/inmanta/execute/proxy.py:24]: Probably need a noqa in here as well.

File: src/inmanta/data/__init__.py [hunk 2]
```diff
@@ -46,25 +46,19 @@
 from asyncpg.protocol import Record
 
 import inmanta.db.versions
+import inmanta.protocol
+import inmanta.types
```
Comment (Hugo-Inmanta) [inline src/inmanta/data/__init__.py:50]: 1. Why do we need protocol in here ?
2. I think we can remove `import inmanta.types` if we import `ResourceType` from `inmanta.types` + change where we use it (only 1 place I think)

File: tests/server/test_incremental_deploy.py [hunk 3]
```diff
@@ -27,16 +27,17 @@
 
 import pytest
 
+import inmanta.types
```
Comment (Hugo-Inmanta) [inline tests/server/test_incremental_deploy.py:30]: I think this isn't required
```suggestion

```

**Review by Hugo-Inmanta — CHANGES_REQUESTED** [review]

**Comment by Hugo-Inmanta** [comment]
Just realized inmanta-lsm will need addressing as well at some point, but should work just fine because we kept things backwards compatible.

File: src/inmanta/execute/proxy.py (hunk 1 above)
Comment (wouterdb) [reply src/inmanta/execute/proxy.py:24]: they are actually used, if it passed, it is fine

**Review by sanderr — APPROVED** [review]

File: src/inmanta/agent/forking_executor.py [hunk 4]
```diff
@@ -97,6 +97,7 @@
 from inmanta.agent import executor, resourcepool
 from inmanta.agent.executor import DeployResult, FactResult
 from inmanta.agent.resourcepool import PoolManager, PoolMember
+from inmanta.const import LOGGER_NAME_EXECUTOR
```
[the remark this answers is not in the document]
Comment (wouterdb) [reply src/inmanta/agent/forking_executor.py:100]: Very much required,.... (i.e. not a change if you look at all commits)

File: src/inmanta/agent/in_process_executor.py [hunk 5]
```diff
@@ -30,7 +30,7 @@
 from inmanta.agent import executor, handler
 from inmanta.agent.executor import DeployResult, DryrunResult, FactResult, FailedResources, ResourceDetails
 from inmanta.agent.handler import HandlerAPI, SkipResource, SkipResourceForDependencies
-from inmanta.const import ParameterSource
+from inmanta.const import NAME_RESOURCE_ACTION_LOGGER, ParameterSource
```
[the remark this answers is not in the document]
Comment (wouterdb) [reply src/inmanta/agent/in_process_executor.py:33]: Very much required,.... (i.e. not a change if you look at all commits)
 (merge from master)

File: src/inmanta/data/__init__.py [hunk 6]
```diff
@@ -60,7 +60,14 @@
 )
 from inmanta.data import model as m
 from inmanta.data import schema
-from inmanta.data.model import AuthMethod, BaseModel, PagingBoundaries, PipConfig, api_boundary_datetime_normalizer
+from inmanta.data.model import (
+    AuthMethod,
+    BaseModel,
+    PagingBoundaries,
+    PipConfig,
+    ResourceType,
```
[the remark this answers is not in the document]
Comment (wouterdb) [reply src/inmanta/data/__init__.py]: doh!

**Review by Hugo-Inmanta — APPROVED** [review]

**Pull request closed without merging**